src/mavlink-router/endpoint.cpp

```diff
@@ -323,11 +323,6 @@ bool Endpoint::accept_msg(int target_sysid, int target_compid, uint8_t src_sysid
         }
     }
 
-    // This endpoint sent the message, we don't want to send it back over the
-    // same channel to avoid loops: reject
-    if (has_sys_comp_id(src_sysid, src_compid))
-        return false;
-
```

Comment:
I think I missed it in the new code: what prevents a loop from happening now?

Reply:
It's in the [mainloop.cpp line 146ff.](https://github.com/intel/mavlink-router/pull/235/files#diff-9e3876545c86b9c8b8f34a888a09d5529acba3e9d937d6f1cadb00e8af3567b2R146). I extended the `Mainloop::route_msg()` method to get the endpoint instance, which received the message. When looping through all endpoint to offer them the message, I leave this one out.
It should at least have the same behavior as the comment of the deleted code you cited.

Reply:
I just figured out, that this filtering is still needed when using redundant data links. So I added that part to the code again.
(But: This might lead to the communication coming to a halt in some cases. I'm currently working on a separate fix for that.)